File: template-service/src/main/java/com/sensa/templateservice/mapper/TemplateMapper.java
package com.sensa.templateservice.mapper;

import com.sensa.templateservice.dto.TemplateRequest;
import com.sensa.templateservice.dto.TemplateResponse;
import com.sensa.templateservice.entity.Template;
import org.springframework.stereotype.Component;

@Component
public class TemplateMapper {
    public Template mapToEntity(TemplateRequest templateRequest) {
        return Template.builder()
                .clientUsername(templateRequest.clientUsername())
                .title(templateRequest.title())
                .content(templateRequest.content())
                .build();
    }

    public TemplateResponse mapToResponse(Template template) {
        return TemplateResponse.builder()
                .clientUsername(template.getClientUsername())
                .title(template.getTitle())
                .content(template.getContent())
                .build();
    }

    public Template update(TemplateRequest templateRequest, Template template) {
        template.setClientUsername(templateRequest.clientUsername());
        template.setTitle(template.getTitle());
        template.setContent(template.getContent());
        return template;
    }
}

File: template-service/src/main/kotlin/com/sensa/templateservice/mapper/TemplateMapper.kt
package com.sensa.templateservice.mapper

import com.sensa.templateservice.dto.TemplateRequest
import com.sensa.templateservice.dto.TemplateResponse
import com.sensa.templateservice.entity.Template
import org.springframework.stereotype.Component
import java.time.LocalDateTime
import java.util.UUID

@Component
class TemplateMapper {

    fun toEntity(request: TemplateRequest, createdBy: UUID): Template = Template(
        templateName = request.templateName,
        description = request.description,
        channel = request.channel,
        content = request.content,
        createdBy = createdBy
    )

    fun toResponse(entity: Template): TemplateResponse = TemplateResponse(
        id = entity.id,
        templateName = entity.templateName,
        description = entity.description,
        channel = entity.channel,
        content = entity.content,
        createdBy = entity.createdBy,
        createdAt = entity.createdAt,
        updatedAt = entity.updatedAt
    )

    fun updateEntity(existing: Template, request: TemplateRequest) {
        existing.templateName = request.templateName
        existing.description = request.description
        existing.channel = request.channel
        existing.content = request.content
        existing.updatedAt = LocalDateTime.now()
    }
}
